### PR TITLE
Commerce 3885 Implement order date filter on orders dataset display 

### PR DESCRIPTION
**Pull request by luca-pellizzon**
Built on top of https://github.com/liferay-frontend/liferay-portal/pull/549
Wait that one to be merged

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Review by riccardo-alberti — APPROVED** [review]

File: modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+		Calendar calendar = Calendar.getInstance();
+
+		int day = calendar.get(Calendar.DAY_OF_MONTH);
+		int month = calendar.get(Calendar.MONTH);
+		int year = calendar.get(Calendar.YEAR) - 1;
+
+		try {
+			User currentUser = _userService.getCurrentUser();
+
+			SearchContext searchContext = _buildSearchContext(
```
Comment (marco-leo) [inline modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java:83]: @luca-pellizzon can we avoid this query and just use a default?

**Review by marco-leo — CHANGES_REQUESTED** [review]

**Review by dsitu — APPROVED** [review]

File: modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java (hunk 1 above)
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java]: @marco-leo let me know what you think about the latest changes

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-service/src/main/resources/META-INF/custom-sql/default.xml [hunk 2]
```diff
@@ -154,6 +154,16 @@
 				CommerceOrder.modifieddate DESC
 		]]>
 	</sql>
+	<sql id="com.liferay.commerce.service.persistence.CommerceOrderFinder.findFirstCommerceOrderDate">
+		<![CDATA[
+			SELECT
+				MIN(CommerceOrder.orderDate) AS FIRST_DATE
```
Comment (marco-leo) [inline modules/apps/commerce/commerce-service/src/main/resources/META-INF/custom-sql/default.xml:160]: @luca-pellizzon This query is expensire, as is without filters and context, even companies

File: modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	enabled = false, immediate = true,
+	property = "clay.data.set.display.name=" + CommerceOrderDataSetConstants.COMMERCE_DATA_SET_KEY_ALL_ORDERS,
+	service = ClayDataSetFilter.class
+)
+public class CommerceOrderCreateDateClayTableDataSetFilter
+	extends BaseDateRangeClayDataSetFilter {
```
Comment (dsitu) [inline modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java:39]: I think we should name this `CommerceOrderCreateDateRangeClayDataSetFilter` to match the class that is being extended.

**Review by dsitu — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-service/src/main/resources/META-INF/custom-sql/default.xml (hunk 2 above)
Comment (riccardo-alberti) [reply modules/apps/commerce/commerce-service/src/main/resources/META-INF/custom-sql/default.xml:160]: we should definitely filter by companyId. but as a note is it not possible to set the min date for the filter to minium possible date (1/1/1970) and then preselect today's date - 1 month as fromValue in the filter? @luca-pellizzon @FabioDiegoMastrorilli

File: modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java (hunk 3 above)
Comment (riccardo-alberti) [reply modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java:39]: also if we filter on orderDate we should add the mapping in the order entity model in the admin-order-api

**Review by riccardo-alberti — CHANGES_REQUESTED** [review]

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Review by marco-leo — APPROVED** [review]

**Comment by riccardo-alberti** [comment]
ci:forward

**Pull request closed without merging**